src/main/java/com/launchdarkly/client/LDClient.java

```diff
@@ -306,21 +303,21 @@ public String stringVariation(String featureKey, LDUser user, String defaultValu
    * @param defaultValue the default value of the flag
    * @return the variation for the given user, or {@code defaultValue} if the flag is disabled in the LaunchDarkly control panel
    */
+  @Override
   public JsonElement jsonVariation(String featureKey, LDUser user, JsonElement defaultValue) {
-    if (isOffline()) {
-      return defaultValue;
-    }
-    JsonElement value = evaluate(featureKey, user, defaultValue);
+    JsonElement value = evaluate(featureKey, user, defaultValue, VariationType.Json);
     return value;
   }
 
-  private JsonElement evaluate(String featureKey, LDUser user, JsonElement defaultValue) {
-    if (user == null || user.getKeyAsString().isEmpty()) {
-      logger.warn("Null user or null/empty user key when evaluating flag: " + featureKey + "; returning default value");
+  private JsonElement evaluate(String featureKey, LDUser user, JsonElement defaultValue, VariationType expectedType) {
+    if (user == null || user.getKey() == null) {
+      logger.warn("Null user or null user key when evaluating flag: " + featureKey + "; returning default value");
       sendFlagRequestEvent(featureKey, user, defaultValue, defaultValue, null);
       return defaultValue;
     }
-
+    if (user.getKeyAsString().isEmpty()) {
```

Comment:
I think we should point this over to a knowledge base item @justinucd that explains the problem in more detail, and change the message, as the implications are bigger than just the user not being stored.

"User key is blank. LaunchDarkly does not support blank user keys-- see [...] for more details"